web/src/components/product/ProductPage.jsx

```diff
@@ -0,0 +1,435 @@
+  const [isDeregistrationPopupOpen, setIsDeregistrationPopupOpen] = useState(false);
+  const [isWarningPopupOpen, setIsWarningPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsDeregistrationPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Deregister")}
+      </Button>
+      <Button
+        variant="secondary"
+        onClick={() => setIsWarningPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductDeregistrationPopup
+        isOpen={isDeregistrationPopupOpen}
+        onFinish={() => setIsDeregistrationPopupOpen(false)}
+        onCancel={() => setIsDeregistrationPopupOpen(false)}
+      />
+      <RegisteredWarningPopup
+        isOpen={isWarningPopupOpen}
+        onAccept={() => setIsWarningPopupOpen(false)}
+      />
+    </>
+  );
+};
+
+/**
+ * Renders the actions for the current product.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const ProductActions = ({ isDisabled = false }) => {
+  const { registration } = useProduct();
+
+  const withRegistration = registration.requirement !== "not-required";
+  const registered = registration.code !== null;
+
+  return (
+    <>
+      <div className="split">
+        <If
+          condition={withRegistration}
+          then={
+            <If
+              condition={registered}
+              then={<RegisteredButtons isDisabled={isDisabled} />}
+              else={<DeregisteredButtons isDisabled={isDisabled} />}
+            />
+          }
+          else={<WithoutRegistrationButtons isDisabled={isDisabled} />}
+        />
+      </div>
+      <If
+        condition={isDisabled}
+        then={
+          <p>{_("Configuring product. Actions are disabled until the product is configured.")}</p>
+        }
+      />
+    </>
+  );
+};
+
+/**
+ * Page for configuring a product.
+ * @component
+ */
+export default function ProductPage() {
+  const [managerStatus, setManagerStatus] = useState();
+  const [softwareStatus, setSoftwareStatus] = useState();
+  const { cancellablePromise } = useCancellablePromise();
+  const { manager, software } = useInstallerClient();
+  const { selectedProduct, registration } = useProduct();
+
+  useEffect(() => {
+    cancellablePromise(manager.getStatus()).then(setManagerStatus);
+    return manager.onStatusChange(setManagerStatus);
+  }, [cancellablePromise, manager]);
+
+  useEffect(() => {
+    cancellablePromise(software.getStatus()).then(setSoftwareStatus);
+    return software.onStatusChange(setSoftwareStatus);
+  }, [cancellablePromise, software]);
+
+  const isLoading = managerStatus === BUSY || softwareStatus === BUSY;
+
+  return (
+    <Page title={_("Product")} icon="inventory_2" actionLabel={_("Back")} actionVariant="secondary">
+      <Section title={selectedProduct.name} loading={isLoading}>
```

Comment:
Question: Is selectedProduct always a thing? I mean, did you use `selectedProduct.name` instead `selectedProduct?.name` on purpuose?

Reply:
Fixed.